public/pages/Dashboard/Container/Dashboard.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import React, { useEffect, useCallback, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppState } from '../../../redux/reducers';
+import { getDetectorList } from '../../../redux/reducers/ad';
+import { SORT_DIRECTION } from '../../../../server/utils/constants';
+import { EmptyDashboard } from '../Components/EmptyDashboard/EmptyDashboard';
+import {
+  EuiPageHeaderSection,
+  EuiTitle,
+  EuiPageHeader,
+  EuiPage,
+} from '@elastic/eui';
+
+export const Dashboard = () => {
+  const dispatch = useDispatch();
+
+  // useCallBack ensures we don't recreate the funciton
+  const onRefreshPage = useCallback(
+    () =>
+      dispatch(
+        getDetectorList({
+          from: 0,
+          size: 1,
+          search: '',
+          sortDirection: SORT_DIRECTION.DESC,
+          sortField: 'name',
+        })
+      ),
+    []
+  );
+
+  const totalDetectors = useSelector(
+    (state: AppState) => state.ad.totalDetectors
+  );
+
+  useEffect(() => {
+    onRefreshPage();
+  });
+
+  return totalDetectors == 0 ? (
```

Comment:
You may also have loading state, else this will trigger flash for user.
Show lower in case of `totalDetector ===0 && isLoading == true`

Reply:
good idea.  changed to show Spanner while loading.